### ISSUE TITLE
Send a notification to a user, when another user saves their post

### ISSUE THREAD

**Issue by PhilipHow** [forem/forem#19510]
## Is this feature related to a problem? Please describe.

As a user, I want to be notified when another user saves one of my posts to see who is enjoying my content and better understand my community.

## Describe the solution you’d like

- In forem#19356, it was discovered that there is no current design for the notification when a user's post has been bookmarked by another user. We want to formalise and implement this design.
- The notification should display the post title and link to the post that has been saved.
- The notification will "collapse" if multiple users save the post on the same day. A collapsed notification will show:
  - Up to two profile images
  - Up to two names (the rest will be summarized, e.g. Philip How, Peter Frank and 3 others saved your post ...)
  - A users profile image and name will provide a link to the user's profile
  - When summarized, the list of users will be sorted by the timestamp of when the post was saved in descending order, so the one or two most recent users who have displayed the article are shown.
  
**Designs for `Save` Notification** - [Figma](https://www.figma.com/file/9UfSjNYJs745w3Q8JCUqsL/Small-Wins---%E2%80%A8Core-Experience-%E2%9A%A1%EF%B8%8F?type=design&node-id=2039-6287&t=6bSTxzUv4lz0mGdz-4)

![Final](https://github.com/forem/forem/assets/84626054/0ef6afbb-5e59-4409-bf05-12e0ebd2885a)

## Definition of Done

- [ ] Notification is sent to users when another user saves their post
- [ ] Notification shows the post title, and links to the post
- [ ] Notification shows at most once per day, per post
- [ ] Collapse notification shows the two most recent users who have saved the post
- [ ] Specs to test new behavior

## Additional Context

* Could be based on the existing reaction notification:
 ![image](https://github.com/forem/forem/assets/1628871/2e3f149f-29a6-4c31-91c6-0371977904aa)

**Comment by filleduchaos** [comment]
@anujbhavsar96 @PhilipHow right now for other aggregated notifications (using reactions for example) the pluralization pattern we use is:

- One reaction: `Anuj Bhavsar reacted to Some Cool Post`
- Two reactions: `Brooklyn Simmons and Anuj Bhavsar reacted to Some Cool Post`
- More than two reactions: `Brooklyn Simmons and 2 others reacted to Some Cool Post`

Are we okay with that being inconsistent with these new notifications?

**Comment by anujbhavsar96** [comment]
@filleduchaos Yes. The larger plan is to display two names first instead of one only and then make it `and 5 others` . We will make similar changes for reactions as well.

One reaction: `Anuj Bhavsar reacted to Some Cool Post`

Two reactions: `Brooklyn Simmons and Anuj Bhavsar reacted to Some Cool Post` _(I didn't add `and` between names before, changing it in design now)_

More than two reactions: `Brooklyn Simmons, Anuj Bhavsar and 2 others reacted to Some Cool Post`

**Comment by filleduchaos** [comment]
@anujbhavsar96 That makes sense! @mirie I think this is good to go as is ✨
